tests/st/calicoctl/test_upgrade.py

```diff
@@ -0,0 +1,610 @@
+# Copyright (c) 2015-2017 Tigera, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import logging
+import os
+import random
+from subprocess import CalledProcessError
+
+import yaml
+from nose_parameterized import parameterized
+
+from tests.st.utils.utils import calicoctl, \
+    name, wipe_etcd, get_ip, clean_calico_data, log_and_run, CalicoctlOutput
+
+ETCD_SCHEME = os.environ.get("ETCD_SCHEME", "http")
+ETCD_CA = os.environ.get("ETCD_CA_CERT_FILE", "")
+ETCD_CERT = os.environ.get("ETCD_CERT_FILE", "")
+ETCD_KEY = os.environ.get("ETCD_KEY_FILE", "")
+ETCD_HOSTNAME_SSL = "etcd-authority-ssl"
+
+logging.basicConfig(level=logging.DEBUG, format="%(message)s")
+logger = logging.getLogger(__name__)
+
+data = {}
```

Comment:
how about moving all the data into https://github.com/projectcalico/calicoctl/blob/master/tests/st/utils/data.py 
like we've done for CRUD test data, maybe call this one v1_data.py or something? WDYT?